### ISSUE TITLE
An item with the same key has already been added. Key: scope

### ISSUE THREAD

**Issue by jochenjonc** [manuel-guilbault/AspNetCore.NonInteractiveOidcHandlers#2]
When caching is not enabled, the following error occurs: An item with the same key has already been added. Key: scope

The cause is the following line:

https://github.com/manuel-guilbault/AspNetCore.NonInteractiveOidcHandlers/blob/a7b74cc41fcee777619213dc4aeaa327d20a5dea/src/AspNetCore.NonInteractiveOidcHandlers/ClientCredentialsTokenHandler.cs#L71

My configuration uses:
```csharp
options.ExtraTokenParameters = new Dictionary<string, string> {{"audience", Configuration.Audience}};
```

The issue is that the Dictionary is reused and thus the second time, keys that already part of the dictionary are added again.

The solution is to create a new dictionary based on the dictionary in the options.

```csharp
Parameters = _options.ExtraTokenParameters ? new Dictionary<string, string>(_options.ExtraTokenParameters) :  new Dictionary<string, string>(),
```

**Comment by manuel-guilbault** [comment]
Hi @jochenjonc, thanks for raising this issue and for your pull request.
I just published [version 1.0.1-pre.2](https://www.nuget.org/packages/AspNetCore.NonInteractiveOidcHandlers/1.0.1-pre.2) of the Nuget package. Can you update and confirm that everything is ok on your end?

**Comment by jochenjonc** [comment]
Hi @manuel-guilbault , the pre-release works as expected.

**Comment by manuel-guilbault** [comment]
@jochenjonc I just published version [1.0.1](https://www.nuget.org/packages/AspNetCore.NonInteractiveOidcHandlers/1.0.1).  Thanks again for your contribution.
